src/core/DatetimeClasses.ts

```diff
@@ -273,22 +314,24 @@ export class HzLocalDateTime {
  */
 export class HzOffsetDateTime {
 
-    hzLocalDateTime: HzLocalDateTime;
     private static readonly timezoneRegex = /([Zz]|[+-]\d\d:\d\d)/;
 
+    constructor(private readonly hzLocalDateTime: HzLocalDateTime, private readonly offsetSeconds: number) {
+    }
+
     /**
+     * Constructs a new instance from Date and offset seconds.
      * @param date Must be a valid date. So `date.getTime()` should be not NaN
      * @param offsetSeconds Must be between -64800-64800 (-+18:00)
      */
-    constructor(date: Date, readonly offsetSeconds: number) {
+    static fromDate(date: Date, offsetSeconds: number): HzOffsetDateTime {
         if (!(date instanceof Date) || isNaN(date.getTime())) {
             throw new RangeError('Invalid date.');
         }
         if (!Number.isInteger(offsetSeconds) || !(offsetSeconds >= -64800 && offsetSeconds <= 64800)) {
             throw new RangeError('Offset seconds can be between -64800(-18:00) and 64800(+18:00).');
         }
-
-        this.hzLocalDateTime = new HzLocalDateTime(
+        const hzLocalDateTime = new HzLocalDateTime(
```

Comment:
We can directly call `HzLocalDateTime.fromDate` here, instead of constructing the object field by field